### PR TITLE
HEP: add support toolkit

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**
Add a support toolkit to improve the support efficiency

**Solution:**
Add support toolkit

**Related Issue:**
https://github.com/harvester/harvester/issues/4935

**Test plan:**
No need

### CODE REVIEW

**Review by Yu-Jack — APPROVED** [review]
Overall, I think it's a good idea. 

I think we can combine this HEP and the ideals of https://github.com/harvester/harvester/issues/8436 and https://github.com/harvester/harvester/issues/6564 to make a comprehensive tool to analyze the cluster.

The picture in my mind is like:

- Some rules are used in admission webhook to prevent users from doing something wrong, like deleting default resources.
- Some rules are used in toolkit to detect, like the VM migration failed due to cd-rom VM during upgrades.
- A service health dashboard showing potential risk, like pre-check. Then users will know that before upgrading.

In short, there are two parts we can do

- Before the event - pre-check + admission webhook
- After the event- support toolkit

I think most rules are shared between these two parts.

**Comment by Yu-Jack** [comment]
BTW, before implementing any features, we should have a discussion about which issues we want to prevent. Listing the tire 1 issues might help us get rid of a lot of issues from customers. I think this list will be a handbook for all team members.
